apps/cowswap-frontend/src/common/pure/OrderProgressBarV2/index.tsx

```diff
@@ -347,6 +442,51 @@ function FinishedStep({
   const isCustomRecipient = order && getIsCustomRecipient(order)
   const receiver = order?.receiver || order?.owner
 
+  const theme = useTheme()
+
+  // Randomly select a benefit message on component initialization
+  const [randomBenefit] = useState(() => COW_SWAP_BENEFITS[Math.floor(Math.random() * COW_SWAP_BENEFITS.length)])
```

Comment:
Nitpick: we have something similar for this https://github.com/cowprotocol/cowswap/blob/1cdfa24c6448e3ebf2c6e3c986cb5d7bfd269aa4/apps/cowswap-frontend/src/modules/fortune/hooks/useOpenRandomFortune.ts#L14

And, do we need to use state for this?
Maybe useMemo instead?

Reply:
Thank you addressed this one just now.